aptos-move/aptos-vm/src/data_cache.rs

```diff
@@ -329,6 +329,15 @@ impl<'e, E: ExecutorView> TDelayedFieldView for StorageAdapter<'e, E> {
         self.executor_view
             .get_reads_needing_exchange(delayed_write_set_keys, skip)
     }
+
+    fn get_group_reads_needing_exchange(
+        &self,
+        delayed_write_set_keys: &HashSet<Self::Identifier>,
+        skip: &HashSet<Self::ResourceKey>,
```

Comment:
@igor-aptos: I understood the flow in the squash (and why skip was used there to filter the writes), but don't follow the interface here, as the block executor also has access to writes? If we do need it, changing to Iter might also be better, but in any case, nothing to do necessarily in this PR.

Reply:
Does creating a `to_skip

Reply:
Oh, I meant the above for the other place where it is a HashMap

It is still faster to pass &, but code wise might be cleaner to pass lambda.